daemon/networkdriver/bridge/driver.go

```diff
@@ -614,27 +599,21 @@ func Release(job *engine.Job) error {
 			logrus.Infof("Unable to release IPv6 %s", err)
 		}
 	}
-	return nil
 }
 
 // Allocate an external port and map it to the interface
-func AllocatePort(job *engine.Job) error {
+func AllocatePort(id string, port nat.Port, binding nat.PortBinding) (nat.PortBinding, error) {
 	var (
-		err error
-
 		ip            = defaultBindingIP
-		id            = job.Args[0]
-		hostIP        = job.Getenv("HostIP")
-		hostPort      = job.GetenvInt("HostPort")
-		containerPort = job.GetenvInt("ContainerPort")
-		proto         = job.Getenv("Proto")
+		proto         = port.Proto()
+		containerPort = port.Int()
```

Comment:
@LK4D4 this here would panic without my change in nat/nat.go. It is replacing `GetenvInt("ContainerPort")`, which was returning 0 in case ParseInt was erroring out: https://github.com/docker/docker/blob/2052d01b1e52a4b1fa79fdc73c52a7b52f6d88a7/engine/env.go#L90